content/master/concepts/composition-functions.md

```diff
@@ -1,1026 +1,699 @@
+* The `status` of any composed resource.
+* The connection details of any composed resource.
 
-        r["resource"]["metadata"]["annotations"][ANNOTATION_KEY_AUTHOR] = author
-        r["resource"]["metadata"]["annotations"][ANNOTATION_KEY_QUOTE] = quote
+A pipeline of functions _accumulates_ desired state. This means that each
+function builds upon the desired state of previous functions in the pipeline.
+Crossplane sends a function the desired state accumulated by all previous
+functions in the pipeline. The function adds to or updates the desired state and
+then passes it on. When the last function in the pipeline has run, Crossplane
+applies the desired state it returns.
 
-    write_Functionio(Functionio)
+{{<hint "important">}}
+A function __must__ copy all desired state from its RunFunctionRequest to its
+RunFunctionResponse. If a function adds a resource to its desired state the next
+function must copy it to its desired state. If it doesn't, Crossplane doesn't
+apply the resource. If the resource exists, Crossplane deletes it.
 
+A function can _intentionally_ choose not to copy parts of the desired state.
+For example a function may choose not to copy a desired resource to prevent that
+resource from existing.
 
-if __name__ == "__main__":
-    main()
-```
+Most function SDKs handle copying desired state automatically.
+{{</hint>}}
 
-Building this function requires its `requirements.txt` and a `Dockerfile`:
-
-{{< expand "The Function's requirements" >}}
-```python
-certifi==2022.12.7
-charset-normalizer==3.0.1
-click==8.1.3
-idna==3.4
-pathspec==0.10.3
-platformdirs==2.6.2
-PyYAML==6.0
-requests==2.28.2
-tomli==2.0.1
-urllib3==1.26.14
-```
-{{< /expand >}}
-
-{{< expand "The Function's Dockerfile" >}}
-```Dockerfile
-FROM debian:11-slim AS build
-RUN apt-get update && \
-    apt-get install --no-install-suggests --no-install-recommends --yes python3-venv && \
-    python3 -m venv /venv && \
-    /venv/bin/pip install --upgrade pip setuptools wheel
-
-FROM build AS build-venv
-COPY requirements.txt /requirements.txt
-RUN /venv/bin/pip install --disable-pip-version-check -r /requirements.txt
-
-FROM gcr.io/distroless/python3-debian11
-COPY --from=build-venv /venv /venv
-COPY . /app
-WORKDIR /app
-ENTRYPOINT ["/venv/bin/python3", "main.py"]
-```
-{{< /expand >}}
+A function should only add the fields it cares about to the desired state. It
+should add these fields every time Crossplane calls it. If a function adds a
+field to the desired state once, but doesn't add it the next time it's called,
+Crossplane deletes the field. The same is true for composed resources. If a
+function adds a composed resource to the desired state, but doesn't add it the
+next time it's called, Crossplane deletes the composed resource.
 
-Create and push the Function just like you would any Docker image.
+{{<hint "tip">}}
+Crossplane uses
+[server side apply](https://kubernetes.io/docs/reference/using-api/apply/)
```

Comment:
This URL gives me a 404.
Was it meant to be this?
https://kubernetes.io/docs/reference/using-api/server-side-apply/

Reply:
It was - thanks! No idea what happened there.